Tumusime14/Certificate-Manager---David
# Feature/Task-6-KAN-64 Supplier search lookup

### PR Description

This PR integrates the `SupplierLookupModal` into the `CertificateForm` component. It allows users to open a modal, search for suppliers, and select a supplier, which then populates the supplier input field in the form.

### JIRA TICKET
[JIRA Ticket](https://task-force-pro-dccs.atlassian.net/jira/software/projects/KAN/boards/1?selectedIssue=KAN-64)
### Checklist

- [x] I have linked related issues or tasks.



## Changes Made

- Added `SupplierLookupModal` component.
- Updated `CertificateForm` to open the modal on clicking the search icon.
- Added functionality to handle supplier selection and close the modal.

## How to test

1. Open the `CertificateForm` component.
2. Click the search icon next to the supplier input field to open the `SupplierLookupModal`.
3. Search for a supplier using the input field in the modal.
4. Select a supplier from the list to populate the supplier input field in the form.
5. Ensure the modal closes automatically after selection.
6. Verify that the selected supplier is displayed in the supplier input field.
7. Test form submission with the populated supplier data to ensure it saves correctly.

### Screenshot
![image](https://github.com/user-attachments/assets/204fb3d3-1226-4d61-b0cc-07865756af88)
